sdk/deviceregistry/azure-resourcemanager-deviceregistry/src/test/java/com/azure/resourcemanager/deviceregistry/DeviceRegistryManagerTests.java

```diff
@@ -0,0 +1,83 @@
+import com.azure.core.management.profile.AzureProfile;
+import com.azure.core.test.TestProxyTestBase;
+import com.azure.core.test.annotation.LiveOnly;
+import com.azure.core.util.Configuration;
+import com.azure.core.util.CoreUtils;
+import com.azure.identity.AzurePowerShellCredentialBuilder;
+import com.azure.resourcemanager.deviceregistry.models.Asset;
+import com.azure.resourcemanager.resources.ResourceManager;
+import com.azure.resourcemanager.resources.fluentcore.policy.ProviderRegistrationPolicy;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+import java.util.Random;
+import java.util.stream.Collectors;
+
+public class DeviceRegistryManagerTests extends TestProxyTestBase {
+    private static final Random RANDOM = new Random();
+    private static final Region REGION = Region.US_EAST;
+    private String resourceGroupName = "rg" + randomPadding();
+    private DeviceRegistryManager deviceRegistryManager = null;
+    private ResourceManager resourceManager;
+    private boolean testEnv;
+
+    @Override
+    public void beforeTest() {
+        final TokenCredential credential = new AzurePowerShellCredentialBuilder().build();
+        final AzureProfile profile = new AzureProfile(AzureEnvironment.AZURE);
+
+        resourceManager = ResourceManager.configure()
+            .withLogOptions(new HttpLogOptions().setLogLevel(HttpLogDetailLevel.BASIC))
+            .authenticate(credential, profile)
+            .withDefaultSubscription();
+
+        deviceRegistryManager = DeviceRegistryManager.configure()
+            .withPolicy(new ProviderRegistrationPolicy(resourceManager))
+            .withLogOptions(new HttpLogOptions().setLogLevel(HttpLogDetailLevel.BASIC))
+            .authenticate(credential, profile);
+
+        // use AZURE_RESOURCE_GROUP_NAME if run in LIVE CI
+        String testResourceGroup = Configuration.getGlobalConfiguration().get("AZURE_RESOURCE_GROUP_NAME");
+        testEnv = !CoreUtils.isNullOrEmpty(testResourceGroup);
+        if (testEnv) {
+            resourceGroupName = testResourceGroup;
+        } else {
+            resourceManager.resourceGroups().define(resourceGroupName).withRegion(REGION).create();
+        }
+    }
+
+    @Override
+    protected void afterTest() {
+        if (!testEnv) {
+            resourceManager.resourceGroups().beginDeleteByName(resourceGroupName);
+        }
+    }
+
+    @Test
+    @LiveOnly
+    public void testListByResourceGroup() {
+        // The DeviceRegistry service must be supported by the service Kubernetes cluster with Azure Arc
+        // and Microsoft.ExtendedLocation, but Kubernetes cluster with Azure Arc can only be created by script.
+        // so only add search test
+        List<Asset> assetList = deviceRegistryManager.assets()
+            .listByResourceGroup(resourceGroupName)
+            .stream()
+            .collect(Collectors.toList());
+        Assertions.assertTrue(assetList.isEmpty());
```

Comment:
Can you also add list to
```
discoveredAssets
schemaRegistries
```

Reply:
The `discoveredAssets` and `schemaRegistries` were removed in version 2024-11-01, so test cases cannot be added.
![image](https://github.com/user-attachments/assets/2e25b782-04e8-4856-8f13-cb3cd02bc0db)
![image](https://github.com/user-attachments/assets/e06da17c-caac-4741-b219-c4e8145e1ee3)

Reply:
What about others?
```
    private AssetEndpointProfiles assetEndpointProfiles;
    private BillingContainers billingContainers;
```

Reply:
- `AssetEndpointProfiles` must be supported by `Kubernetes cluster with Azure Arc` and `Microsoft.ExtendedLocation`, and `Kubernetes cluster with Azure Arc` can only be created through scripts.
- `BillingContainers` only supports `get` and `list`.

Reply:
Yes, I mean list. If they support list, do a list too.